### ISSUE TITLE
processDisconnect() nc.err setting has no effect

### ISSUE THREAD

**Issue by mcqueary** [nats-io/nats#116]
Here's `processDisconnect()` from `nats.go`:
```go
func (nc *Conn) processDisconnect() {
	nc.status = DISCONNECTED
	if nc.err != nil {
		return
	}
	if nc.info.TLSRequired {
		nc.err = ErrSecureConnRequired
	} else {
		nc.err = ErrConnectionClosed
	}
}
```

Here's the only place where processDisconnect is called:

```go
func (nc *Conn) processOpErr(err error) {
	nc.mu.Lock()
	if nc.isConnecting() || nc.isClosed() || nc.isReconnecting() {
		nc.mu.Unlock()
		return
	}

	if nc.Opts.AllowReconnect && nc.status == CONNECTED {
		// Set our new status
		nc.status = RECONNECTING

		if nc.ptmr != nil {
			nc.ptmr.Stop()
		}
		if nc.conn != nil {
			nc.bw.Flush()
			nc.conn.Close()
			nc.conn = nil
		}
		go nc.doReconnect()

		nc.mu.Unlock()
		return
	} else {
		nc.processDisconnect()
		nc.err = err
		nc.mu.Unlock()
		nc.Close()
	}
}
```
`nc.err` is immediately reassigned, therefore setting it in processDisconnect() has no effect.

**Comment by derekcollison** [comment]
Feel free to remove and PR if all tests pass.

**Comment by mcqueary** [comment]
Will do, I just wanted to open the issue in case I missed a detail.

On Monday, December 21, 2015, Derek Collison <notifications@github.com>
wrote:

> Feel free to remove and PR if all tests pass.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nats-io/nats/issues/116#issuecomment-166295102>.
>


-- 
*Larry McQueary* | Director, Messaging Technology
larry@apcera.com | @mcqueary | github.com/mcqueary

**Comment by mcqueary** [comment]
Closed by #117
